shared/src/question-generators/math/modularArithmetic/eea.ts

```diff
@@ -0,0 +1,234 @@
+import { Language } from "@shared/api/Language"
+import {
+  MultiFreeTextFeedbackFunction,
+  MultiFreeTextFormatFunction,
+  MultiFreeTextQuestion,
+  QuestionGenerator,
+} from "@shared/api/QuestionGenerator"
+import { serializeGeneratorCall } from "@shared/api/QuestionRouter"
+import Random from "@shared/utils/random"
+import { t, tFunctional, Translations } from "@shared/utils/translations"
+
+const translations: Translations = {
+  en: {
+    name: "Extended Euclidean Algorithm",
+    description:
+      "Fill in the steps to calculate the GCD of two numbers using the Extended Euclidean Algorithm.",
+    eeaQuestion: "Calculate the GCD of ${{a}}$ and ${{b}}$ by completing the following equations:",
+    linearCombinationPrompt: "Express the GCD as a linear combination:",
+    gcd: "gcd",
+    feedbackInvalid: "Input must be an integer.",
+  },
+  de: {
+    name: "Erweiterter Euklidischer Algorithmus",
+    description:
+      "Fülle die Schritte aus, um den GGT zweier Zahlen mit dem erweiterten euklidischen Algorithmus zu berechnen.",
+    eeaQuestion:
+      "Berechne den GGT von ${{a}}$ und ${{b}}$ durch das Vervollständigen der folgenden Gleichungen:",
+    linearCombinationPrompt: "Drücke den GGT als Linearkombination aus:",
+    gcd: "ggt",
+    feedbackInvalid: "Eingabe muss eine ganze Zahl sein.",
+  },
+}
+
+export const ExtendedEuclideanAlgorithm: QuestionGenerator = {
+  id: "eea",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["gcd", "extended euclidean algorithm", "eea"],
+  languages: ["en", "de"],
+  expectedParameters: [{ type: "string", name: "variant", allowedValues: ["default"] }],
+
+  generate: (lang: Language, parameters, seed) => {
+    const random = new Random(seed)
+    const a = random.int(50, 100)
+    const b = random.int(10, 50)
+    const { steps } = calculateEEA(a, b)
+    const permalink = serializeGeneratorCall({
+      generator: ExtendedEuclideanAlgorithm,
+      lang,
+      parameters,
+      seed,
+    })
+
+    const table = generateEEATableSteps(a, b, lang, steps)
+
+    const checkFormat: MultiFreeTextFormatFunction = ({ text }, fieldID) => {
+      if (text[fieldID].trim() === "") return { valid: false, message: "" }
+
+      if (!/^-?\d+$/.test(text[fieldID])) {
```

Comment:
This check is inconsistent with `parseInt` used in the feedback function, although everything passing this test can also be parsed by `parseInt`. However, either use the same function (`parseInt` + check for NaN) or at least allow for leading/trailing whitespaces (using `trim` or `\s` in the regex).